### ISSUE TITLE
Multiple actions per agent in the distributed mode

### ISSUE THREAD

**Issue by kosmylo** [Future-Power-Networks/MAPDN#34]
In the `distributed` mode each agent is responsible to control one generator. In your case one agent has only one action. If I want to have multiple actions per agent what changes should I make?

**Comment by hsvgbkhgbv** [comment]
You need to manually override the following functions: `get_action(self)`, `get_total_actions(self)`, `get_avail_actions(self)`, `get_avail_agent_actions(self, agent_id)` and `get_num_of_agents(self)`

**Comment by kosmylo** [comment]
I was able to identify where the error came from. First, according to the error, the shape of the actions was `[1,1,4]`. However, I would expect `[1,5,4]`. In the function `get_actions(self, state, status, exploration, actions_avail, target=False, last_hid=None)` of `maddpg.py` the variable means had the correct shape `[1, 5, 4]`, but then the problem was due to the following part:

```
if means.size(-1) > 1:
    means_ = means.sum(dim=1, keepdim=True)
    log_stds_ = log_stds.sum(dim=1, keepdim=True)
 else:
    means_ = means
    log_stds_ = log_stds
```

My question now is what is the purpose of this part?
